src/state/resolver.ts

```diff
@@ -13,43 +14,79 @@ export interface IStateResolver {
   resolve(id: bigint, state: bigint): Promise<ResolvedState>;
   rootResolve(state: bigint): Promise<ResolvedState>;
 }
+
 export type ResolvedState = {
   latest: boolean;
   genesis: boolean;
   state: unknown;
   transitionTimestamp: number | string;
 };
-export class EthStateResolver implements IStateResolver {
-  private rpcUrl: string;
-  private contractAddress: string;
 
-  constructor(rpcUrl: string, contractAddress: string) {
-    this.rpcUrl = rpcUrl;
-    this.contractAddress = contractAddress;
-  }
+export type CacheOptions = {
+  stateResolveCache?: ICache<ResolvedState>;
+  rootResolveCache?: ICache<ResolvedState>;
+};
 
-  public async resolve(id: bigint, state: bigint): Promise<ResolvedState> {
-    const url = new URL(this.rpcUrl);
+export class EthStateResolver implements IStateResolver {
+  private _contract: Abi;
+  private _stateResolveCache?: ICache<ResolvedState>;
+  private _rootResolveCache?: ICache<ResolvedState>;
+
+  constructor(rpcUrl: string, contractAddress: string, cacheOptions?: CacheOptions) {
+    const url = new URL(rpcUrl);
     const ethersProvider = new ethers.providers.JsonRpcProvider({
+      skipFetchSetup: true,
```

Comment:
can you please remind the purpose of this option?

Reply:
from doc:
```Normally a connection will specify the default values for a connection such as CORS-behavior and caching policy, to ensure compatibility across platforms. On some services, such as Cloudflare Workers, specifying any value (inclluding the default values) will cause failure. Setting this to true will prevent any values being passed to the underlying API.```

let's pass it as an option to EthStateResolver in case cloudfare rpc is used by specific customer